devito/stencilkernel.py

```diff
@@ -511,6 +518,22 @@ def cfunction(self):
             self._cfunction.argtypes = self.argtypes
 
         return self._cfunction
+   
+    @property
+    def cfunctionJ(self):
```

Comment:
Ok, so the basic idea behind the compilation toolchain is to provide a cascade that can be intercepted at any point - but obviously you require the entry point between compile and load. So, why don't we split up the current cascade to expose that instead of adding another version of `cfunction`? I'm thinking we could turn `basename` into a property and provide a utility function `compile()` that simply does `jit_compile(self.ccode, self.basename, compiler=self.compiler)`. In the above `cfunction` we can then replace `jit_compile_and_load()` with
```
self.compile()
self.lib = load(self.basename, compiler=self.compiler)
```